src/Raven.Server/RavenServer.cs

```diff
@@ -1076,11 +1086,18 @@ protected async Task<byte[]> RefreshViaLetsEncrypt(CertificateHolder currentCert
                         await ServerStore.SendToLeaderAsync(new RecheckStatusOfServerCertificateCommand());
                         return null;
                     }
-
+                    
                     if (certUpdate.TryGet(nameof(CertificateReplacement.Replaced), out int replaced) == false)
                         replaced = 0;
 
-                    if (nodesInCluster > replaced)
+                    if (certUpdate.TryGet(nameof(CertificateReplacement.ReplacedNodes), out BlittableJsonReaderArray replacedNodesBlittable) == false)
+                    {
+                        throw new InvalidOperationException($"Expected to get '{nameof(CertificateReplacement.ReplacedNodes)}'");
+                    }
+
+                    var replacedNodes = replacedNodesBlittable.Items.Select(n => n.ToString()).ToHashSet();
+                    if ((replacedNodes.Contains("*") && replaced < nodesInCluster.Count) || 
```

Comment:
I think this might result in a stuck replacement process.

Assume 2 cluster nodes, one new, one old.
I'm and old node, which got confirmation from a new one, I will not place its tag anywhere, but increase the confirmations locally.

Now, this old node before he execute his own confirmation got updated to the newer version.. and only then sends the confirmation command _with_ a tag. (because it is new)

we will end up with one node having both tags, while the other will have only one tag, without the '*'.
Not sure what will happened afterwards.

I think we can't rely on the '*' and need to do an 'or' for both logic.

Reply:
Changed